src/power_grid_model_io/converters/pandapower_converter.py

```diff
@@ -0,0 +1,361 @@
+        self._create_pgm_input_sym_gens()
+        self._create_pgm_input_three_winding_transformers()
+
+    def _create_pgm_input_nodes(self):
+        assert "node" not in self.pgm_data
+
+        pp_busses = self.pp_data["bus"]
+        pgm_nodes = initialize_array(data_type="input", component_type="node", shape=len(pp_busses))
+        pgm_nodes["id"] = self._generate_ids("bus", pp_busses.index)
+        pgm_nodes["u_rated"] = pp_busses["vn_kv"] * 1e3
+
+        self.pgm_data["node"] = pgm_nodes
+
+    def _create_pgm_input_lines(self):
+        assert "line" not in self.pgm_data
+
+        pp_lines = self.pp_data["line"]
+
+        # Add an extra column called 'index' containing the index values, which we need to do the join with switches
+        pp_lines["index"] = pp_lines.index
+
+        # Select the appropriate switches and columns
+        pp_switches = self.pp_data["switch"]
+        pp_switches = pp_switches[self.pp_data["switch"]["et"] == "l"]
+        pp_switches = pp_switches[["element", "bus", "closed"]]
+
+        # Join the switches with the lines twice, once for the from_bus and once for the to_bus
+        pp_from_switches = (
+            pp_lines[["index", "from_bus"]]
+            .merge(
+                pp_switches,
+                how="left",
+                left_on=["index", "from_bus"],
+                right_on=["element", "bus"],
+            )
+            .fillna(True)
+            .set_index(pp_lines.index)
+        )
+        pp_to_switches = (
+            pp_lines[["index", "to_bus"]]
+            .merge(
+                pp_switches,
+                how="left",
+                left_on=["index", "to_bus"],
+                right_on=["element", "bus"],
+            )
+            .fillna(True)
+            .set_index(pp_lines.index)
+        )
+
+        pgm_lines = initialize_array(data_type="input", component_type="line", shape=len(pp_lines))
+        pgm_lines["id"] = self._generate_ids("line", pp_lines.index)
+        pgm_lines["from_node"] = self._get_ids("bus", pp_lines["from_bus"])
+        pgm_lines["from_status"] = pp_lines["in_service"] & pp_from_switches["closed"]
+        pgm_lines["to_node"] = self._get_ids("bus", pp_lines["to_bus"])
+        pgm_lines["to_status"] = pp_lines["in_service"] & pp_to_switches["closed"]
+        pgm_lines["r1"] = pp_lines["r_ohm_per_km"] * pp_lines["length_km"]
+        pgm_lines["x1"] = pp_lines["x_ohm_per_km"] * pp_lines["length_km"]
+        pgm_lines["c1"] = pp_lines["c_nf_per_km"] * pp_lines["length_km"]
+        pgm_lines["tan1"] = 0.0
+        pgm_lines["i_n"] = pp_lines["max_i_ka"] * 1e3
+
+        self.pgm_data["line"] = pgm_lines
+
+    def _create_pgm_input_sources(self):
+        assert "source" not in self.pgm_data
+
+        pp_ext_grid = self.pp_data["ext_grid"]
+
+        pgm_sources = initialize_array(data_type="input", component_type="source", shape=len(pp_ext_grid))
+        pgm_sources["id"] = self._generate_ids("source", pp_ext_grid.index)
```

Comment:
I think the ids would be generated by the `table name` ie. names of components from pandapower. Then it be `ext_grid` here and similar in other functions as well.

Reply:
Yes, that is true, thanks! I assume that we can't do that with loads, because we create three of them. I do leave them as "sym_load_const_power", "sym_load_const_impedance", etc. Right?

Reply:
Oh, your next comment regards loads